gldemo/plugin.cpp

```diff
@@ -320,20 +348,23 @@ class gldemo : public threadloop {
 	// This may be changed later, but it really doesn't matter for this purpose because
 	// it will be replaced by a real, Monado-interfaced application.
 	virtual void start() override {
-		glXMakeCurrent(xwin->dpy, xwin->win, xwin->glc);
+		assert(errno == 0 && "Errno should not be set at start of gldemo start function");
 
-		// Initialize the GLFW library, still need it to get time
-		if(!glfwInit()){
-			printf("Failed to initialize glfw\n");
+		if (!glXMakeCurrent(xwin->dpy, xwin->win, xwin->glc)) {
+			std::cerr << "glXMakeCurrent returned false in " << __FILE__ << ":" << __LINE__ << std::endl;
```

Comment:
I think it's best to `assert(0 && "something bad happened")`. It's less typing anyway.

Unfortunately, `assert(glXstuff() ...)` (eliding the if-statement) won't work because it won't run the expression when `profile = opt`.

Reply:
Agreed. The value can be stored temporarily.